### PR TITLE
Add size support for strings

### PR DESCRIPTION
**Pull request by urswiss**
Replace `integer` field type by `string` field type, as `integer` type is nowhere used (`int` is used instead) and the `@Size` annotation is not supported on integers anyway, see https://www.baeldung.com/javax-validation

### CODE REVIEW

File: gradle-plugin/src/main/groovy/ch/silviowangler/gradle/restapi/PluginTypes.java [hunk 1]
```diff
@@ -31,9 +31,6 @@
 public enum PluginTypes {
 
 	RESTAPI_IDTYPE(ClassName.get("ch.silviowangler.rest.types", "IdType")),
-	RESTAPI_FILTERMODEL(ClassName.get("ch.silviowangler.rest.model", "FilterModel")),
```
Comment (saw303) [inline gradle-plugin/src/main/groovy/ch/silviowangler/gradle/restapi/PluginTypes.java]: The removal of these constants is not part of the PR. Please revert the changes on this file

File: gradle-plugin/src/main/groovy/ch/silviowangler/gradle/restapi/builder/AbstractResourceBuilder.java [hunk 2]
```diff
@@ -373,11 +373,13 @@ protected String getHttpMethod() {
 					Number min = field.getMin();
 					Number max = field.getMax();
 
-					if (field.getType().equalsIgnoreCase("integer")) {
```
Comment (saw303) [inline gradle-plugin/src/main/groovy/ch/silviowangler/gradle/restapi/builder/AbstractResourceBuilder.java]: I agree that `integer` min/max validation seems to be an untested and unused feature. But why remove it instead of fix it by adding `@Min` and `@Max` annotations?

File: gradle-plugin/src/main/groovy/ch/silviowangler/gradle/restapi/builder/AbstractResourceBuilder.java [hunk 3]
```diff
@@ -373,11 +373,13 @@ protected String getHttpMethod() {
...
+					if (field.getType().equalsIgnoreCase("string")) {
+						int minInt = min.intValue();
+						int maxInt = max.intValue();
 						fieldBuilder.addAnnotation(
 								AnnotationSpec.builder(JAVAX_VALIDATION_SIZE.getClassName())
-										.addMember("min", "$N", min.intValue())
-										.addMember("max", "$N", max.intValue()).build()
+										.addMember("min", "$L", String.valueOf(minInt))
```
Comment (saw303) [inline gradle-plugin/src/main/groovy/ch/silviowangler/gradle/restapi/builder/AbstractResourceBuilder.java]: `String.valueOf()` is not needed

**Comment by urswiss** [comment]
this PR is on hold because it's not clear what to do with optional fields, how to avoid that the `@Size` annotation will also check null values?
